storage/namespace.go

```diff
@@ -522,7 +564,23 @@ func (n *dbNamespace) Flush(
 	return res
 }
 
-func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
+func (n *dbNamespace) NeedsFlush(startInclusive time.Time, endInclusive time.Time) bool {
+	// don't need to flush for bad input
+	if startInclusive.After(endInclusive) {
+		return false
+	}
+
+	// find all blockStarts for the given range
+	var (
+		blockSize   = n.nopts.RetentionOptions().BlockSize()
+		startBlock  = startInclusive.Truncate(blockSize)
+		endBlock    = endInclusive.Truncate(blockSize).Add(blockSize)
+		blockStarts []time.Time
+	)
+	for t := startBlock; t.Before(endBlock); t = t.Add(blockSize) {
+		blockStarts = append(blockStarts, t)
+	}
```

Comment:
Seems like we'll do this in different places in the code base, should we create a method to create the relevant block starts?